cpp/oneapi/dal/algo/pca/backend/gpu/infer_kernel_dpc.cpp

```diff
@@ -26,39 +26,156 @@
 namespace oneapi::dal::pca::backend {
 
 namespace pr = oneapi::dal::backend::primitives;
-
+namespace bk = oneapi::dal::backend;
 using dal::backend::context_gpu;
+using alloc = sycl::usm::alloc;
+
 using model_t = model<task::dim_reduction>;
 using input_t = infer_input<task::dim_reduction>;
 using result_t = infer_result<task::dim_reduction>;
 using descriptor_t = detail::descriptor_base<task::dim_reduction>;
 
+template <typename Float>
+auto get_centered(sycl::queue& q,
+                  pr::ndview<Float, 2>& data,
+                  const pr::ndview<Float, 1>& means,
+                  const bk::event_vector& deps = {}) {
+    ONEDAL_PROFILER_TASK(compute_centered_data, q);
+    const std::int64_t row_count = data.get_dimension(0);
+    const std::int64_t column_count = data.get_dimension(1);
+
+    auto centered_data_ptr = data.get_mutable_data();
+    auto means_ptr = means.get_data();
+
+    auto centered_event = q.submit([&](sycl::handler& h) {
+        const auto range = bk::make_range_2d(row_count, column_count);
+        h.depends_on(deps);
+        h.parallel_for(range, [=](sycl::id<2> id) {
+            const std::int64_t i = id[0];
+            const std::int64_t j = id[1];
```

Comment:
Don't think we need this implicit conversions. It is perfectly fine to use `std::size_t` in array addressing